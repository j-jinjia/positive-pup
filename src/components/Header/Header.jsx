import "./Header.scss";

const Header = (props) => {
  const { headingText, subheadingText, isLeftAlign } = props;
  let headerClassName = "header";
  if (isLeftAlign) headerClassName += " header--left-align";

  return (
    <div className={headerClassName}>
      <h1 className="header__heading"> {headingText} </h1>
      <h2 className="header__subheading">{subheadingText}</h2>
    </div>
  );
};

export default Header;
