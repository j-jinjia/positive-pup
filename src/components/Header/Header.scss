@use "../../styles/variables" as *;

.header {
  text-align: center;
  padding-left: 2rem;
  padding-right: 2rem;

  &__heading {
    color: $colour-primary;
    font-size: $font-size-h1;
    font-family: $font-header;
    margin-bottom: 0.5rem;
    text-transform: uppercase;
  }

  &__subheading {
    font-size: $font-size-medium;
    font-family: $font-header;
    font-weight: normal;
    margin-bottom: 1.75rem;
  }

  &--left-align {
    text-align: left;
    padding: 0;
  }
}

@media screen and (min-width: $desktop-breakpoint) {
  .header {
    &__subheading {
      font-size: $font-size-large;
    }
  }
}
