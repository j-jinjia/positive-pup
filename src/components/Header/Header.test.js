import { customRender } from "../../utils/testUtils";
import Header from "./Header";

it("should match the Header component snapshot", () => {
  const { container } = customRender(
    <Header
      headingText={"About Us"}
      subheadingText={"headerJSX"}
      subheadingSecondLine={"My name is Melanie!"}
      isLeftAlign={true}
    />
  );

  expect(container).toMatchSnapshot();
});
