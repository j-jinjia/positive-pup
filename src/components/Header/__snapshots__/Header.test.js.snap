// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the Header component snapshot 1`] = `
<div>
  <div
    class="headerheader"
  >
    <h1
      class="header__heading"
    >
       
       
    </h1>
    <h2
      class="header__subheading"
    />
  </div>
</div>
`;
